### ISSUE TITLE
SO3 rotmat -> quat canonicalization changes downstream quaternion trajectories

### ISSUE THREAD

**Issue by abcamiletto** [abcamiletto/nanomanifold#33]
I isolated a second downstream parity drift to `SO3.convert(..., src="rotmat", dst="quat")` while simplifying a motion-generation codebase.

Context:
- Downstream task: Kimodo text/constraint-conditioned motion generation
- Framework: PyTorch
- Comparison: end-to-end output tensors against a saved upstream reference
- Metric: `max_abs_diff`

Observed behavior when swapping only `rotmat -> quat` into the pipeline:
- worst case: `1.1798888444900513e-03`
- worst field/case: `unconditional / local_rot_mats`
- other cases: roughly `1.6e-6` to `4.8e-6`

Control measurements from the same run:
- `quat -> rotmat` alone: `<= 5.96e-7`
- `axis_angle -> rotmat` alone: `0.0`
- `matrix -> sixd` alone: `0.0`

So this looks specifically tied to `from_rotmat(...)->to_quat(...)` rather than quaternion handling in general.

Current `nanomanifold` path in `nanomanifold/SO3/primitives/rotmat.py` ends with:
```python
w = xp.where(cond1, w1, xp.where(cond2, w2, xp.where(cond3, w3, w4)))
x = xp.where(cond1, x1, xp.where(cond2, x2, xp.where(cond3, x3, x4)))
y = xp.where(cond1, y1, xp.where(cond2, y2, xp.where(cond3, y3, y4)))
z = xp.where(cond1, z1, xp.where(cond2, z2, xp.where(cond3, z3, z4)))

quat = canonicalize(xp.stack([w, x, y, z], axis=-1), xp=xp)
return to_quat(quat, convention=convention, xp=xp)
```

Reference implementation from the downstream codebase:
```python
def _matrix_to_quaternion(matrix: torch.Tensor) -> torch.Tensor:
    if matrix.size(-1) != 3 or matrix.size(-2) != 3:
        raise ValueError(f"Invalid rotation matrix shape {matrix.shape}.")

    batch_dim = matrix.shape[:-2]
    m00, m01, m02, m10, m11, m12, m20, m21, m22 = torch.unbind(matrix.reshape(batch_dim + (9,)), dim=-1)
    q_abs = _sqrt_positive_part(
        torch.stack(
            [
                1.0 + m00 + m11 + m22,
                1.0 + m00 - m11 - m22,
                1.0 - m00 + m11 - m22,
                1.0 - m00 - m11 + m22,
            ],
            dim=-1,
        )
    )
    quat_by_rijk = torch.stack(
        [
            torch.stack([q_abs[..., 0] ** 2, m21 - m12, m02 - m20, m10 - m01], dim=-1),
            torch.stack([m21 - m12, q_abs[..., 1] ** 2, m10 + m01, m02 + m20], dim=-1),
            torch.stack([m02 - m20, m10 + m01, q_abs[..., 2] ** 2, m12 + m21], dim=-1),
            torch.stack([m10 - m01, m20 + m02, m21 + m12, q_abs[..., 3] ** 2], dim=-1),
        ],
        dim=-2,
    )
    floor = torch.tensor(0.1, device=q_abs.device, dtype=q_abs.dtype)
    quat_candidates = quat_by_rijk / (2.0 * q_abs[..., None].max(floor))
    choice = F.one_hot(q_abs.argmax(dim=-1), num_classes=4)[..., None]
    return (choice * quat_candidates).sum(dim=-2).reshape(batch_dim + (4,))
```

The important differences are:
- `nanomanifold` computes a branch with `cond1/cond2/cond3`, then canonicalizes the quaternion.
- the downstream implementation computes four candidates and selects the one associated with the largest `q_abs`.
- `nanomanifold` normalizes/canonicalizes the output quaternion, effectively enforcing a sign convention.
- the downstream implementation does not canonicalize the sign, which preserves the branch implied by the input matrix sequence.

That sign/branch choice matters here because the downstream postprocess consumes quaternion trajectories directly, not just the represented rotations.

Feature request:
- could `nanomanifold` expose a `rotmat -> quat` path without canonicalization, or with configurable canonicalization/sign strategy?
- alternatively, if there is already a supported way to preserve the raw branch choice implied by the input matrices, a documented example would help.

I can provide a minimal downstream repro if useful.
